Journaley/Utilities/PostMarkdownParser.cs

```diff
@@ -0,0 +1,114 @@
+        /// to Journaley.
+        /// </summary>
+        /// <param name="formattedString">Formatted HTML string</param>
+        /// <returns>Properly formatted HTML string for publishing.</returns>
+        public static string PostMarkdown(string formattedString)
+        {
+            // First convert all code blocks using <p> into <pre>
+            // which is used in almost all implementations of
+            // Markdown, even in the original Markdown specs.
+            formattedString = Regex.Replace(formattedString, @"<p><code>", "<pre><code>", RegexOptions.Multiline);
+            formattedString = Regex.Replace(formattedString, @"</code></p>", "</code></pre>", RegexOptions.Multiline);
+
+            System.Text.StringBuilder builder = new System.Text.StringBuilder();
+            System.Text.StringBuilder paragraphBuilder = new System.Text.StringBuilder();
+
+            var isNewLine = 0;
+
+            string newLineString;
+            string strippedOverString;
+
+            var lines = formattedString.Split(
+                new string[] { "\r\n", "\n" },
+                StringSplitOptions.None);
+
+            for (int i = 0; i < lines.Length - 1; ++i)
+            {
+                string line = lines[i];
+
+                // We first incrementally check if the line begins with <p> tag. When it does,
+                // We turn on the paragraphBuilder to append the remaining lines.
+
+                // If the checks hit a line with the </p> closing, we disable check and proceed
+                // to parse the text, add <br /> tags and then adding it to builder
+                // where it is assembled with the rest of the file.
+                
+                // We dump the text to builder if it hits a <pre> tag.
+                if (line.Contains("<p>"))
+                {
+                    isNewLine = 1;
+                }
+                else if (line.Contains("</p>"))
+                {
+                    isNewLine = 0;
+                }
+                else if (line.Contains("<pre>"))
+                {
+                    isNewLine = -1;
+                }
+
+                if (isNewLine == 1)
+                {
+                    paragraphBuilder.AppendLine(line);
+
+                    if (line.Contains("</p>"))
+                    {
+                        // Single line strikethroughs.
+                        line = Regex.Replace(line, @"(~~)(.*?)(~~)", "<del>$2</del>", RegexOptions.Multiline);
+                        builder.AppendLine(line);
+                        paragraphBuilder.Clear();
+                        isNewLine = 0;
+                    }
+                }
+                else if (isNewLine == 0)
+                {
+                    paragraphBuilder.AppendLine(line);
+
+                    newLineString = paragraphBuilder.ToString();
+
+                    // Multiline strikethrough.
+                    if (Regex.IsMatch(newLineString, @"(~~)(.*?)(~~)"))
```

Comment:
Yeah, I think you're right about that.

I actually assumed that problem, because supposedly in Day one you can do this:
```
~~Nikki~~
Nikki
~~Nikki~~
```
And it parses it correctly like:
~~Nikki~~
Nikki
~~Nikki~~

But in the previous code, it parsed like:

~~Nikki~~ Nikki
Nikki

That's why I added an regex check, but I think because of the foreach, it might have fixed it.